pkg/cpi/commonUtils.go

```diff
@@ -40,7 +40,11 @@ func (tokenParameters TokenParameters) GetBearerToken() (string, error) {
 	if httpErr != nil {
 		return "", errors.Wrapf(httpErr, "HTTP %v request to %v failed with error", method, tokenFinalURL)
 	}
-	defer resp.Body.Close()
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
```

Comment:
Same as above, first defer, then check the error, then check if response is nil.